demo/ARContainerViewController.swift

```diff
@@ -92,9 +89,29 @@ final class ARContainerViewController: UIViewController {
 }
 
 extension ARContainerViewController: NavigationServiceDelegate {
-    
+
     func navigationService(_ service: NavigationService, didUpdate progress: RouteProgress, with location: CLLocation, rawLocation: CLLocation) {
         instructionsLabel.isHidden = false
         instructionsLabel.text = progress.currentLegProgress.currentStep.instructions
+
+        if routeHasChanged {
+            routeHasChanged = false
+            navigationDelegate?.navigationService(didUpdate: Route(route: progress.route))
+        }
+
+        if progress.currentLegProgress.userHasArrivedAtWaypoint {
+            routeHasChanged = true
+            navigationDelegate?.navigationServiceArrivedAtDestination()
```

Comment:
Looks like we reached only a waypoint, not a destination.
Could you please explain why we call `navigationServiceArrivedAtDestination`?

Reply:
Not sure about implementation of `navigationService(_ service: NavigationService, didArriveAt waypoint: Waypoint ` at first sight, probably you will want to discuss the logic behind.

Reply:
Isn't waypoint a destination?

Reply:
done